village/src/App.js

```diff
@@ -12,13 +15,79 @@ class App extends Component {
     };
   }
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:3333/smurfs')
+      .then(res => {
+        this.setState({ smurfs: res.data });
+      })
+      .catch(err => {
+        console.log(err);
+        this.setState({ error: err });
+      })
+  }
+
+  addSmurf = (smurf) => {
+    
+    axios
+      .post("http://localhost:3333/smurfs", smurf)
+      .then(res => {
+        this.setState({ smurfs: res.data });
+      })
+      .catch(err => {
+        console.log(err);
+        this.setState({ error: err });
+       
+      })
+  }
```

Comment:
Good job adding any errors to state. How might you take advantage of that and show the user that error if it exists?